api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +21,49 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// [#next-free-field: 7]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message Route {
+    // String database name that must exact match with client connection db name
+    string database = 1;
+
+    // Upstream cluster to forward the command to.
+    string cluster = 2;
+  }
+
+  message ConnectionPoolSettings {
+    // Max connections exists in the connection pool, default is 10. FIXME(qinggniq)
+    google.protobuf.UInt32Value max_connections = 1;
+
+    // Max spare connections in the connection pool, addition spare connection will be closed, default is 0.
+    google.protobuf.UInt32Value max_idle_connections = 2;
+
+    // Connections when envoy setup, default is 0.
+    google.protobuf.UInt32Value start_connections = 3;
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  repeated Route routes = 3;
+
+  ConnectionPoolSettings setting = 4 [(validate.rules).message = {required: true}];
+
+  config.core.v3.DataSource downstream_auth_username = 5 [(udpa.annotations.sensitive) = true];
+
+  config.core.v3.DataSource downstream_auth_password = 6 [(udpa.annotations.sensitive) = true];
+}
+
+message MySQLProtocolOptions {
+  config.core.v3.DataSource auth_password = 1 [(udpa.annotations.sensitive) = true];
```

Comment:
Suggest a shared message for username/password.

Reply:
Do you mean remove `udpa.annotations.sensitive = true`? I just found at redis username/password is secret at https://github.com/envoyproxy/envoy/blob/a240824c376693b16ca8be51e435b95c42f3449f/api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto#L308-L318, I don't know is there any differences between them.

Reply:
I'm saying grouping username/password with senstitivity options into a single message is good and makes sense als for downstream above; the same username/password message can be reused for both upstream/downstream.